src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5385,7 +5385,7 @@ and TcAdjustExprForTypeDirectedConversions (cenv: cenv) (overallTy: OverallTy) a
     let g = cenv.g
 
     match overallTy with
-    | MustConvertTo (_, reqdTy) when g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions ->
+    | MustConvertTo (isMethodArg, reqdTy) when g.langVersion.SupportsFeature LanguageFeature.AdditionalTypeDirectedConversions || (g.langVersion.SupportsFeature LanguageFeature.NullableOptionalInterop && isMethodArg) ->
```

Comment:
Thinking aloud: The conditional here is now quite complex - maybe it would have been better originally to just always call `AdjustExprForTypeDirectedConversions` and have the conditions on the specific rules in that method.  

No need to change anything here though.